pkg/resource/api.go

```diff
@@ -124,19 +126,15 @@ func (a *APIManagedConnectionPropagator) PropagateConnection(ctx context.Context
 		}
 		cmcs.Data = mgcs.Data
 		meta.AddAnnotations(cmcs, map[string]string{
-			AnnotationKeyPropagateFromNamespace: mgcs.GetNamespace(),
-			AnnotationKeyPropagateFromName:      mgcs.GetName(),
-			AnnotationKeyPropagateFromUID:       string(mgcs.GetUID()),
+			fmt.Sprintf(AnnotationKeyPropagateFromFormat, string(mgcs.GetUID())): strings.Join([]string{mgcs.GetNamespace(), mgcs.GetName()}, "/"),
```

Comment:
Note: we do not have any mechanism that restricts a propagated `Secret` to only be propagated from one source. We could theoretically encounter a situation where a `Secret` had multiple `from.propagate.crossplane.io/` annotations and controllers could fight to update the secret's data. We could check here that the `Secret` does not already have a propagate from annotation and refuse to operate on it if so. We could also remove any existing and replace with the new annotation. Otherwise, we could just let this unexpected scenario happen and leave it to a user to fix the conflict.

Reply:
Yes! I was thinking and I believe there are two steps to ensuring this:

1. In `Secret` propagation here we could refuse to add from annotations to a `Secret` that already has them. This is the only place where the annotations are actually set so unless annotations were manually applied by a user, this would restrict from ever having more than one set. Alternatively, we could remove any existing from annotations here and replace them with the desired set. However, this could cause issues if two controllers are both reconciling and removing each others from annotations over and over.
2. We could refuse to operate on `Secrets` that have more than one set of from annotations in the secret propagating reconciler, essentially refusing to fight over updating a `Secret`. This would mean that any `Secret` that somehow obtained multiple sets of from annotations would not have data propagated from either. Because we are restricting the ability to add multiple sets of from annotations via propagation in (1) this situation would only be encountered if a user manually add additional from annotations.

Thoughts?

Reply:
This sounds good to me. Perhaps given we'll only ever allow one propagate-from annotation we should maintain the existing annotation style in which the from UID is the value, not part of the key? I could go either way - I think the existing pattern would make it more obvious that [there can be only one](https://www.youtube.com/watch?v=_J3VeogFUOs) from annotation, but I it would make the to and from annotations asymmetrical.

Reply:
I agree that we should go with the existing pattern. I like that it is asymmetrical with the to annotations because they behave differently :)

Reply:
Using the existing pattern of `from` annotations actually solves both (1) and (2) above :)